### PR TITLE
Added e2e testing

### PR DESCRIPTION
**Pull request by SimonSimCity**
Added a foundation of e2e testing. Please give feedback!

How variables can be set can be found here: https://docs.cypress.io/guides/guides/environment-variables#Setting

I've added the two currently in use (`USERNAME` and `PASSWORD`) already to the github-script which should already work correctly after setting 'em up.

The `build_and_test` job fails because of the missing credentials.

We should also address here if we want the project to deploy if the e2e tests are failing due to a shift in data (e.g. test-data is not available as expected) or if the login service is unavailable or doesn't let us pass for other reasons.

### CODE REVIEW

**Comment by kkuepper** [comment]
I added the 2 secrets.
But the build fails in the linter step

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
```
[the remark this answers is not in the document]
Comment (SimonSimCity) [reply cypress/support/commands.ts:1]: I haven't yet fully understood what the code-examples here are for - they're provided by the code generator. I'd commit it as part of understanding of what we can do, but I don't have any strong opinion on this ... What would you say is better?

Btw, the first line is a so-called [triple-slash directive](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-types-)

**Comment by SimonSimCity** [comment]
It seems like within github-actions, we have a problem reaching out to `localhost` as described in https://github.com/cypress-io/github-action/issues/802 ... Testing out different strategies to see which may get it to work without switching from the name `localhost` ... Or we'll have to change the addresses allowed by for the auth0 credentials.

**Comment by SimonSimCity** [comment]
Problem referenced previously (regarding `localhost`) isn't relevant here. After watching the video of the test, I saw that there's a problem with the test-user we have, which should be solved by our login service.

Videos and screenshots of tests are from now on available as artifacts on this project (!to everyone!). Steps for typing in credentials or confidential data should have the option `log` set to `false`: https://stackoverflow.com/questions/50568526/is-there-any-way-to-hide-a-log-in-cypress

**Review by kkuepper — APPROVED** [review]

**Pull request merged**